### ISSUE TITLE
Trouble running doctests

### ISSUE THREAD

**Issue by ThomasA** [epfl-lts2/pyunlocbox#12]
When I run the test suite locally, i.e. run `make test` in the repo root, I get a lot of problems with doctests failing. One type of problem is with recognizing output NumPy arrays. For example:

    ----------------------------------------------------------------------
    File "pyunlocbox/acceleration.py", line 298, in acceleration.py
    Failed example:
        ret['sol']
    Expected:
        array([4.0002509 , 5.00031362, 6.00037635, 7.00043907])
    Got:
        array([ 4.0002509 ,  5.00031362,  6.00037635,  7.00043907])
    ----------------------------------------------------------------------

Is anyone else seeing this? I have observed this under Ubuntu Linux running Anaconda's Python both with Python 2.7.11 + NumPy 1.11.0 and with Python 3.6.2 +  NumPy 1.13.1.

The above problem can be fixed with `doctest`'s `NORMALIZE_WHITESPACE` option:
``` python
>>> ret['sol']  # doctest: +NORMALIZE_WHITESPACE
    array([ 4., 5., 6., 7.])
```
instead of:
``` python
>>> ret['sol']
    array([4., 5., 6., 7.])
```
I am also seeing another, related problem. Apparently the following test problem does not solve in the same number of iterations (anymore / on my computer) as when the test was written:

    ----------------------------------------------------------------------
    File "pyunlocbox/acceleration.py", line 193, in acceleration.py
    Failed example:
        ret = solvers.solve([f1, f2], x0, solver, atol=1e-32, rtol=None)
    Expected:
        Solution found after 4 iterations:
            objective function f(sol) = 0.000000e+00
            stopping criterion: ATOL
    Got:
        Solution found after 1 iterations:
            objective function f(sol) = 0.000000e+00
            stopping criterion: ATOL
    ----------------------------------------------------------------------

This problem can be fixed with `doctest`'s `ELLIPSIS` option:
``` python
>>> ret = solvers.solve([f1, f2], x0, solver, atol=1e-32, rtol=None)  # doctest: +ELLIPSIS
    Solution found after ... iterations:
        objective function f(sol) = 0.000000e+00
        stopping criterion: ATOL
```
instead of:
```python
>>> ret = solvers.solve([f1, f2], x0, solver, atol=1e-32, rtol=None)
    Solution found after 4 iterations:
        objective function f(sol) = 0.000000e+00
        stopping criterion: ATOL
```
Please advise what should be done here. Maybe it is important to test that the problem is solved in exactly 4 iterations?

**Comment by rodrigo-pena** [comment]
Hi @ThomasA , I was the one who implemented the acceleration module. As far as the number of iterations is concerned, it is not important that this particular example be solved in exactly 4 iterations. The guarantees of the acceleration scheme are not that precise. I had just hardcoded the number 4 because I did nott know about those doctest options 😄 

I remember I got frustrated with the mismatched spaces in my local machine as well. I think the doctest options are elegant. What do you say, @mdeff ?

**Comment by ThomasA** [comment]
Maybe some of these `doctest` options are worth setting globally?

**Comment by mdeff** [comment]
The mismatched spaces issue is due to numpy which has changed its default printing behavior (in version 1.14 if I remember well). Update your numpy to the latest version and they should go away.

Good idea to use ellipsis for the number of iterations, I'm all for it. :) Then I think we won't need that much of them to set them globally.

**Comment by ThomasA** [comment]
Could it be an idea to normalize whitespace anyway to avoid this problem in the future?

**Comment by mdeff** [comment]
I'm not sure. For one I hope they won't change it again... Then, may we miss legitimate errors if we normalize?

**Comment by ThomasA** [comment]
I'm not sure about that. If you keep it to array comparisons, whitespace should not matter - only numerical values. On the other hand you could just ignore it for now and deal with it if they change it again.

**Comment by mdeff** [comment]
Agreed. I guess for now it's just easier to stay with the second option. To be reconsidered if / when the formatting changes. :)

**Comment by ThomasA** [comment]
I can confirm that the array comparison problem goes away with NumPy 1.14.
Should I open a PR to fix the iterations issue?

**Comment by mdeff** [comment]
Thanks! Please do :)

**Comment by ThomasA** [comment]
Should I just fix only the problematic test in 'acceleration.py' mentioned above for now? Might it reveal regressions if the iteration count changes in other examples like this one?

**Comment by mdeff** [comment]
I would just fix the problematic one for now. Iteration counts should be deterministic if we seed the random generator when choosing the starting point (that's what I usually do).

**Comment by mdeff** [comment]
Except if @rodrigo-pena sees other places where iteration counts could change.

**Comment by mdeff** [comment]
Addressed in #14
